### PR TITLE
SPUI-449 > Add banners scroll to account and market

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/app/screens/market/view/components/market_banner/market_banner.dart [hunk 1]
```diff
@@ -3,14 +3,13 @@ import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:simple_kit/simple_kit.dart';
 
+import '../../../../../../service/services/signal_r/model/campaign_response_model.dart';
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/market_banner/market_banner.dart:6]: Why this widget is so complex?
you can just simply do like this:

```dart
class MarketBanners extends HookWidget {
  const MarketBanners({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final state = useProvider(campaignNotipod(true));
    final notifier = useProvider(campaignNotipod(true).notifier);
    final colors = useProvider(sColorPod);
    final deepLinkService = useProvider(deepLinkServicePod);

    final controller = PageController(viewportFraction: 0.88);

    if (state.isNotEmpty) {
      return SizedBox(
        height: 120,
        child: PageView.builder(
          controller: controller,
          itemCount: state.length,
          itemBuilder: (_, index) {
            final campaign = state[index];

            return GestureDetector(
              onTap: () {
                deepLinkService.handle(
                  Uri.parse(campaign.deepLink),
                );
              },
              child: SRewardBanner(
                color: setBannerColor(index, colors),
                primaryText: campaign.title,
                imageUrl: campaign.imageUrl,
                primaryTextStyle: sTextH5Style,
                onClose: () {
                  notifier.deleteCampaign(campaign);
                },
                indentRight: state.length == 1,
              ),
            );
          },
        ),
      );
    } else {
      return const SizedBox();
    }
  }
}
```

File: lib/app/screens/market/view/components/market_banner/market_banner.dart [hunk 2]
```diff
@@ -3,14 +3,13 @@ import 'package:flutter_hooks/flutter_hooks.dart';
...
 import '../../../../../../shared/providers/deep_link_service_pod.dart';
 import '../../../../../shared/features/rewards/notifier/campaign_notipod.dart';
 import '../../../../../shared/helpers/set_banner_colors.dart';
 
 class MarketBanner extends HookWidget {
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/market_banner/market_banner.dart:11]: rename to MarketBanners because it's a set of widgets

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import 'package:flutter/material.dart';
-
-import '../../../simple_kit.dart';
-import '../../colors/view/simple_colors_light.dart';
+import 'helper/create_banners_list.dart';
+import 'helper/set_sized_banner_height.dart';
```
[the remark this answers is not in the document]
Comment (DenisMartych) [reply plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart:3]: done

File: plugins/simple_kit/lib/src/account/banners/helper/set_sized_banner_height.dart [hunk 4]
```diff
@@ -0,0 +1,13 @@
+double setSizedBoxHeight({
```
[the remark this answers is not in the document]
Comment (DenisMartych) [reply plugins/simple_kit/lib/src/account/banners/helper/set_sized_banner_height.dart:1]: done

File: plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart [hunk 5]
```diff
@@ -21,6 +20,21 @@ class SimpleAccountBannerList extends StatelessWidget {
...
+
+    final banners = createBannersList(
+      twoFaEnabled: twoFaEnabled,
+      kycPassed: kycPassed,
+      phoneVerified: phoneVerified,
+      onChatBannerTap: onChatBannerTap,
+      onTwoFaBannerTap: onTwoFaBannerTap,
+    );
+
+    final pages = List.generate(
```
[the remark this answers is not in the document]
Comment (DenisMartych) [reply plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart:33]: remove

File: plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart [hunk 6]
```diff
@@ -1,7 +1,6 @@
 import 'package:flutter/material.dart';
-
-import '../../../simple_kit.dart';
-import '../../colors/view/simple_colors_light.dart';
+import 'helper/create_banners_list.dart';
+import 'helper/set_sized_banner_height.dart';
 
 class SimpleAccountBannerList extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (DenisMartych) [reply plugins/simple_kit/lib/src/account/banners/simple_account_banner_list.dart:5]: done

File: lib/app/screens/market/view/components/market_banner/market_banner.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/app/screens/market/view/components/market_banner/market_banner.dart:11]: done

**Pull request closed without merging**